server/server.go

```diff
@@ -42,72 +55,362 @@ func HandleWebsocketConnection(serverHTTPResponse http.ResponseWriter, clientHTT
+	// Initialize an empty Response structure
+	messageFromClient.Response = &backendProtobuf.BackendMessage_Response{}
+	// Initialize an empty Message structure
+	messageToClient := backendProtobuf.BackendMessage{}
+	// Initialize an empty Request structure
+	messageToClient.Request = &backendProtobuf.BackendMessage_Request{}
+	// Initialize an empty Auth structure
+	messageToClient.Request.Auth = &backendProtobuf.BackendMessage_Auth{}
+	// Create a byte slice to store our random bytes
+	backendRandomBytes := make([]byte, 4)
+	// Read random bytes into our slice
+	_, randomBytesErr := rand.Read(backendRandomBytes)
+	// If there is an error reading random bytes
+	if randomBytesErr != nil {
+		return "", randomBytesErr
+	} // if randomBytesErr != nil
+	// Set the byte sequence that the client needs to sign
+	messageToClient.Request.Auth.ToSign = backendRandomBytes
 	// Prepare the data to be sent to the client
-	protobufAuthBytes, protobufAuthBytesErr := proto.Marshal(&protobufAuth)
-	if protobufAuthBytesErr != nil {
+	messageToClientBytes, messageToClientBytesErr := proto.Marshal(&messageToClient)
+	if messageToClientBytesErr != nil {
 		// If there is an error while trying to perform protobuf marshaling, terminate the connection
-		return websocket.BinaryMessage, protobufAuthBytesErr
-	}
+		return "", messageToClientBytesErr
+	} // if messageToClientBytesErr != nil
 	// Send the protobuf data to the client containing the sequence of bytes he needs to sign
-	websocketConnection.WriteMessage(websocket.BinaryMessage, protobufAuthBytes)
+	websocketConnection.WriteMessage(websocket.BinaryMessage, messageToClientBytes)
 	// Read the response from the client which should contain his IdenetityPublicKey and the signed byte sequence
-	_, clientMessage, readMessageErr := websocketConnection.ReadMessage()
+	_, messageFromClientProto, readMessageErr := websocketConnection.ReadMessage()
 	if readMessageErr != nil {
 		// If there is an error while reading the response from the client, terminate the connection
-		return websocket.BinaryMessage, readMessageErr
+		return "", readMessageErr
 	}
 	// Unmarshal the response from the client into our protobuf Auth structure
-	protoUnmarshalErr := proto.Unmarshal(clientMessage, &protobufAuth)
+	protoUnmarshalErr := proto.Unmarshal(messageFromClientProto, &messageFromClient)
 	if protoUnmarshalErr != nil {
 		// If there is an error while unmarshaling the response from the client, terminate the connection
-		return websocket.BinaryMessage, protoUnmarshalErr
+		return "", protoUnmarshalErr
 	}
 	// Create a [32]byte{} identityPublicKey to satisfy ed25519.Verify() type requirements
 	identityPublicKey := [32]byte{}
 	// Create a [64]byte{} signedByteSequenceFromClient to satisfy ed25519.Verify() type requirements
 	signedByteSequenceFromClient := [64]byte{}
+	// Create a string representation of the Identity Public Key
+	identityPublicKeyString := string(messageFromClient.Response.Auth.IdentityPublicKey)
```

Comment:
if you want to "stringify" a public key I would always recommend to use `hex.EncodeToString` (or how ever the method is called). It encodes the bytes into a hex string.

Reply:
Yes, as mentioned in the other comments related to the same suggestions, I was already sending hexEncoded representation from the client code that I wrote for the testing.

Somehow it feels safer if the client is already transmitting hex encoded data, I have no problem changing it though.

Do you think the backend should be doing the hexEncoding?